File: frontend/src/services/brandService.ts
// src/services/brandService.ts

import axiosInstance from "./axiosInstance";
import { Brand } from "../interfaces/Brand";

export const getAllBrands = () => axiosInstance.get<Brand[]>("/brands");
export const getBrandById = (id: number) => axiosInstance.get<Brand>(`/brands/${id}`);
export const createBrand = (brand: Brand) => axiosInstance.post<Brand>("/brands", brand);
export const updateBrand = (id: number, brand: Brand) =>
  axiosInstance.put<Brand>(`/brands/${id}`, brand);
export const deleteBrand = (id: number) => axiosInstance.delete(`/brands/${id}`);

File: backend/src/main/java/com/example/backend/service/impliments/BrandService.java
package com.example.backend.service.impliments;


import com.example.backend.persistence.dao.BrandRepository;
import com.example.backend.persistence.entities.Brand;
import com.example.backend.service.interfaces.IBrand;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class BrandService implements IBrand {

    private final BrandRepository brandRepository;

    @Override
    public List<Brand> getAllBrands() {
        return brandRepository.findAll();
    }

    @Override
    public Brand saveBrand(Brand brand) {
        return brandRepository.save(brand);
    }

    @Override
    public Optional<Brand> getBrandById(Long id) {
        return brandRepository.findById(id);
    }

    @Override
    public void deleteBrand(Long id) {
        brandRepository.deleteById(id);
    }
}
